app/views/play/cinematic/CinematicLankBoss.js

```diff
@@ -17,164 +37,289 @@ const Lank = require('lib/surface/Lank')
       } else {
-        this.moveLank('left', position)
+        commands.push(this.moveLank({ key: side, resource, pos }))
       }
     }
 
+    const background = getBackground(shot)
+    if (background) {
+      commands.push(this.setBackgroundCommand(background))
+    }
+
+    const lHero = getLeftHero(shot)
+    const original = me.get('heroConfig').thangType
+    if (lHero) {
+      const { enterOnStart, thang: { pos } } = lHero
+      moveCharacter('left', original, enterOnStart, pos)
+    }
+
+    const rHero = getRightHero(shot)
+    if (rHero) {
+      const { enterOnStart, thang: { pos } } = rHero
+      moveCharacter('right', original, enterOnStart, pos)
+    }
+
+    const leftCharSlug = getLeftCharacterThangTypeSlug(shot)
+    if (leftCharSlug) {
+      const { slug, enterOnStart, thang: { pos } } = leftCharSlug
+      moveCharacter('left', slug, enterOnStart, pos)
+    }
+
     const rightCharSlug = getRightCharacterThangTypeSlug(shot)
     if (rightCharSlug) {
-      const { slug, enterOnStart, position } = rightCharSlug
-      this.addLank('right', this.loader.getThangType(slug))
-      if (enterOnStart) {
-        commands.push(this.moveLankCommand('right', position))
-      } else {
-        this.moveLank('right', position)
-      }
+      const { slug, enterOnStart, thang: { pos } } = rightCharSlug
+      moveCharacter('right', slug, enterOnStart, pos)
     }
 
     return commands
   }
 
-  registerLank (side, lank) {
-    assertSide(side)
-    this[side] = lank
+  parseDialogNode (dialogNode) {
+    const commands = []
+    const char = exitCharacter(dialogNode)
+    if (char === 'left' || char === 'both') {
+      commands.push(this.moveLankCommand({ key: 'left', pos: { x: -20 } }))
+    }
+
+    if (char === 'right' || char === 'both') {
+      commands.push(this.moveLankCommand({ key: 'right', pos: { x: 20 } }))
+    }
+
+    const bgObject = getBackgroundObject(dialogNode)
+    if (bgObject) {
+      const { scaleX, scaleY, pos: { x, y }, type: { slug } } = bgObject
+      const thangOptions = {
+        scaleFactorX: scaleX,
+        scaleFactorY: scaleY,
+        pos: { x, y },
+        stateChanged: true
+      }
+      commands.push(this.moveLank({ key: BACKGROUND_OBJECT, resource: slug, pos: { x, y }, thang: createThang(thangOptions) }))
+    }
+
+    const removeBgDelay = getClearBackgroundObject(dialogNode)
+    if (typeof removeBgDelay === 'number') {
+      commands.push(new SequentialCommands([
+        new Sleep(removeBgDelay),
+        new SyncFunction(() => {
+          this.removeLank(BACKGROUND_OBJECT)
+        })
+      ]))
+    }
+    return commands
   }
 
   /**
    * Moves either the left or right lank to a given co-ordinates **instantly**.
    * @param {'left'|'right'} side - the lank being moved.
    * @param {{x, y}} pos - the position in meters to move towards.
    */
-  moveLank (side, pos = {}) {
-    assertSide(side)
+  moveLank ({ key, resource, pos = {}, thang }) {
```

Comment:
How is `moveLank` and `moveLankCommand` different? They both appear to return commands.
Maybe rename this to `moveLankInstantCommand` and update jsDoc?

Reply:
Could rename this one to `moveLankInstant`, but it kinda seems like we should consolidate them.  Is this one the same as moveLankCommand with ms=0?

Reply:
It's like ms=0 but without the implementation bloat of the animation command.... hmmm. You're right! Implementation details should be concealed in the method! I could check if ms is zero and handle appropriately in the moveLankCommand method.